client/js/widgets/widget.js

```diff
@@ -244,11 +249,8 @@ export class Widget extends StateManaged {
   }
 
   applyZ(force) {
-    const thisInheritChildZ = this.get('inheritChildZ');
-    if(force || thisInheritChildZ) {
+    if(force || this.get('inheritChildZ')) {
       this.domElement.style.zIndex = this.calculateZ();
-      if(thisInheritChildZ && this.get('parent'))
```

Comment:
It seems like this should have been checking for a parent with `inheritChildZ`, and updating the parent if applyZ was called.